src/components/AllSurveysPage.jsx

```diff
@@ -1,11 +1,41 @@
-import React from 'react';
+import Immutable from 'immutable';
+import React, { PropTypes } from 'react';
 import PureComponent from 'react-pure-render/component';
 
+import { ALL_SURVEYS_PAGE, NEW_SURVEY_PAGE } from '../constants';
+import { formPath } from '../utils.js';
+
 export default class AllSurveysPage extends PureComponent {
+  static propTypes = {
+    sentSurveys: PropTypes.instanceOf(Immutable.Map).isRequired,
+    unsentSurveys: PropTypes.instanceOf(Immutable.Map).isRequired,
+    fetchSurveysMetadata: PropTypes.func.isRequired,
+  };
+
+  static contextTypes = {
+    router: React.PropTypes.object.isRequired,
+  };
+
+  componentDidMount() {
+    this.props.fetchSurveysMetadata();
+  }
+
   render() {
+    const { sentSurveys, unsentSurveys } = this.props;
+    const { router } = this.context;
+    console.log(sentSurveys);
```

Comment:
Yeah I initially added them for debugging, but I left them as an example for people new to React to see how to use props within a component without  calling `this.props.*` every time.